### PR TITLE
Infra cleanup to avoid produce output assemblies and TimeZoneData project file cleanup

### PR DESCRIPTION
**Pull request by ViktorHofer**
https://github.com/dotnet/runtime-assets/pull/102#issuecomment-736883105.

Also removing the dependency node in the nuget package as we theoretically don't need to restore .NETStandard2.0 to be able to consume the package as it's content only.

### CODE REVIEW

**Review by akoeplinger — APPROVED** [review]

**Review by Newread23 — COMMENTED** [review]
#566 #565 #562

File: Directory.Build.props
Comment (Newread23) [inline Directory.Build.props:1]: @filechanged

File: src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj
Comment (Newread23) [inline src/System.Runtime.TimeZoneData/System.Runtime.TimeZoneData.csproj:1]: @filechanged

**Review by Newread23 — COMMENTED** [review]
#103 @@filechanged

**Pull request merged**